src/Controllers/API/AddCourseController.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Portal\Controllers\API;
+
+use Portal\Abstracts\Controller;
+use Portal\Validators\DateTimeValidator;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Psr\Http\Message\ResponseInterface as Response;
+use Slim\Views\PhpRenderer;
+use Portal\Models\CourseModel;
+use Portal\Entities\CourseEntity;
+
+class AddCourseController extends Controller
+{
+    private $courseModel;
+
+    /**
+     * AddCourseController constructor
+     *
+     * @param CourseModel $courseModel
+     */
+    public function __construct(CourseModel $courseModel)
+    {
+        $this->courseModel = $courseModel;
+    }
+
+    public function __invoke(Request $request, Response $response, array $args)
+    {
+        $newCourse = $request->getParsedBody();
+        $responseData = [
+            'success' => false,
+            'message' => 'Unexpected Error.',
+            'data' => []
+        ];
+        $statusCode = 400;
+
+        try {
+            $validDate = DateTimeValidator::validateStartEndTime($newCourse['startDate'], $newCourse['endDate']);
```

Comment:
I'd suggest moving this validation to the CourseEntity, if you have a look at the applicant entity there is quite a nice example of how to do it. You could then also use the string validator for the other fields